lib/sgpres.gi

```diff
@@ -2766,38 +2766,1163 @@ end);
 ##  augmented coset table data. It must not be applied to augmented coset
 ##  tables which are not of type 2.)
 InstallGlobalFunction(CopiedAugmentedCosetTable,function(aug)
-local t;
-  t:=rec(
-	  isAugmentedCosetTable:=true,
-	  type:=aug.type,
-	  tableType:=aug.tableType,
-	  groupGenerators:=aug.groupGenerators,
-	  groupRelators:=aug.groupRelators,
-	  cosetTable:=aug.cosetTable,
-	  cosetFactorTable:=aug.cosetFactorTable,
-	  primaryGeneratorWords:=aug.primaryGeneratorWords,
-	  tree:=aug.tree,
-	  treeNumbers:=aug.treeNumbers,
-	  numberOfSubgroupGenerators:=aug.numberOfSubgroupGenerators,
-	  nameOfSubgroupGenerators:=aug.nameOfSubgroupGenerators,
-	  subgroupGenerators:=aug.subgroupGenerators
-	);
-  if IsBound(aug.conversionList) then
-    t.conversionList:=aug.conversionList;
+local t,j;
+  if IsBound(aug.isNewAugmentedTable) then
+    t:=rec(isNewAugmentedTable:=true);
+    for j in 
+      [ "A", "aug", "ct", "defcount", "from", "homgenims", "homgens",
+      "index", "n", "offset", "primaryImages", "rels",
+      "secondary", "secount", "secondaryImages", "subgens" ] do
+      if IsBound(aug.(j)) then
+	t.(j):=aug.(j);
+      fi;
+    od;
+  else
+    # old version
+    t:=rec(
+	    isAugmentedCosetTable:=true,
+	    type:=aug.type,
+	    tableType:=aug.tableType,
+	    groupGenerators:=aug.groupGenerators,
+	    groupRelators:=aug.groupRelators,
+	    cosetTable:=aug.cosetTable,
+	    cosetFactorTable:=aug.cosetFactorTable,
+	    primaryGeneratorWords:=aug.primaryGeneratorWords,
+	    tree:=aug.tree,
+	    treeNumbers:=aug.treeNumbers,
+	    numberOfSubgroupGenerators:=aug.numberOfSubgroupGenerators,
+	    nameOfSubgroupGenerators:=aug.nameOfSubgroupGenerators,
+	    subgroupGenerators:=aug.subgroupGenerators
+	  );
+    if IsBound(aug.secondaryWords) then
+      t.secondaryWords:=Immutable(aug.secondaryWords);
+    fi;
+
+    if IsBound(aug.conversionList) then
+      t.conversionList:=aug.conversionList;
+    fi;
+    if IsBound(aug.primarySubgroupGenerators) then
+      t.primarySubgroupGenerators:=Immutable(aug.primarySubgroupGenerators);
+    fi;
+    if IsBound(aug.subgroupRelators) then
+      t.subgroupRelators:=Immutable(aug.subgroupRelators);
+    fi;
+    if IsBound(aug.translationTable) then
+      t.translationTable:=Immutable(aug.translationTable);
+    fi;
+
+  fi;
+  return t;
+end);
+
+
+# New version of MTC based on DFH's book
```

Comment:
So after reading the pull request, I now understand what "DFH's book" refers to. I suggest this new comment:

```
# New implemention of the Modified Todd-Coxeter (MTC) algorithm, based on Chapter 5
# of the "Handbook of Computational Group Theory", by Derek F. Holt (refered to as "Handbook" from here on).
```
Then, later on, we can add additional documentation comments.